core/state/metadata.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+// BalanceChangeReason is used to indicate the reason for a balance change, useful
+// for tracing and reporting.
+type BalanceChangeReason byte
+
+const (
+	BalanceChangeUnspecified          BalanceChangeReason = 0
+	BalanceChangeRewardMineUncle      BalanceChangeReason = 1
+	BalanceChangeRewardMineBlock      BalanceChangeReason = 2
+	BalanceChangeDaoRefundContract    BalanceChangeReason = 3
+	BalanceChangeDaoAdjustBalance     BalanceChangeReason = 4
+	BalanceChangeTransfer             BalanceChangeReason = 5
+	BalanceChangeGenesisBalance       BalanceChangeReason = 6
+	BalanceChangeGasBuy               BalanceChangeReason = 7
+	BalanceChangeRewardTransactionFee BalanceChangeReason = 8
+	BalanceChangeGasRefund            BalanceChangeReason = 9
+	BalanceChangeTouchAccount         BalanceChangeReason = 10
+	// TODO: rename (debit, credit)
+	// BalanceChangeSuicideRefund is added to the recipient as indicated by a selfdestructing account.
+	BalanceChangeSuicideRefund BalanceChangeReason = 11
```

Comment:
`BalanceChange`~~Suicide~~ and ~~Refund~~
So, `refund` is an existing concept in the ethereum whitepaper, and the go-ethereum implementation. It has nothing to do with the selfdestruct operation -- which has not been named `suicide` for several years. 

Als (below) `withdraw` is another concept which already exists, and has nothing to do with `selfdestruct`.

Reply:
Renamed to `BalanceIncreaseSelfdestruct` and `BalanceDecreaseSelfdestruct` as you suggested